OMF_API/AppSettings.cs

```diff
@@ -0,0 +1,119 @@
+        /// The base endpoint. E.g. https://dat-b.osisoft.com for OCS
+        /// </summary>
+        public string Resource { get; set; }
+
+        /// <summary>
+        /// The name of the Namespace in OCS that is being sent to
+        /// </summary>
+        public string NamespaceName { get; set; }
+
+        /// <summary>
+        /// The name of the Tenant ID of the Tenant in OCS that is being sent to
+        /// </summary>
+        public string Tenant { get; set; }
+
+        /// <summary>
+        /// The client ID that is being used for authenticating to OCS
+        /// </summary>
+        public string ClientId { get; set; }
+
+        /// <summary>
+        /// The client secret that is being used for authenticating to OCS
+        /// </summary>
+        public string ClientSecret { get; set; }
+
+        /// <summary>
+        /// The API version of the endpoint
+        /// </summary>
+        public string ApiVersion { get; set; }
+
+        /// <summary>
+        /// A feature flag for verifying SSL when connecting to the endpoint
+        /// </summary>
+        public object VerifySSL { get; set; }
+
+        /// <summary>
+        /// A feature flag for enabling compression on messages send to endpoint
+        /// </summary>
+        public bool UseCompression { get; set; }
+
+        /// <summary>
+        /// An optional timeout setting for web requests
+        /// </summary>
+        public int WebRequestTimeoutSeconds { get; set; }
+
+        /// <summary>
+        /// The name of the PI Data Archive that is being sent to
+        /// </summary>
+        public string DataServerName { get; set; }
+
+        /// <summary>
+        /// The username that is being used for authenticating to the PI Web API
+        /// </summary>
+        public string Username { get; set; }
+
+        /// <summary>
+        /// The password that is being used for authenticating to the PI Web API
+        /// </summary>
+        public string Password { get; set; }
+
+        /// <summary>
+        /// The token used to authenticate to an OCS endpoint
+        /// </summary>
+        public string Token { get; set; }
+
+        /// <summary>
+        /// returns the base endpoint URL of endpoint
+        /// </summary>
+        public string BaseEndpoint
+        {
+            get
+            {
+                string BaseEndpoint = "";
+
+                if (string.Equals(this.EndpointType, "OCS"))
+                {
+                    BaseEndpoint = $"{this.Resource}/api/{this.ApiVersion}/tenants/{this.Tenant}/namespaces/{this.NamespaceName}";
+                }
+                else if (string.Equals(this.EndpointType, "EDS"))
+                {
+                    BaseEndpoint = $"{this.Resource}/api/{this.ApiVersion}/tenants/default/namespaces/default";
+                }
+                else if (string.Equals(this.EndpointType, "PI"))
+                {
+                    BaseEndpoint = this.Resource;
+                }
+
```

Comment:
should we have an error handler here for an endpointtype that isn't listed? It could be a good idea since these are case sensitive comparisons. If enums are used, that would be different. I just imagine someone wondering why their baseendpoint is "" when they listed "ocs" or "eds"

Reply:
There is an overload of string.Equals that accepts another parameter, `StringComparisonType`. So this should be for example:
`string.Equals(this.EndpointType, "OCS", StringComparisonType.OrdinalIgnoreCase)`
But, even then a fallback to throw an error or something if there is no matching expected string might be nice.